galsim/phase_psf.py

```diff
@@ -0,0 +1,1399 @@
+    argument.  If the phase aberrations are significant, however, the PSF image size computed this
+    way may still not be sufficiently large to avoid aliasing.  To further increase the pupil plane
+    sampling rate (and hence the PSF image size), you can increase the value of the `pad_factor`
+    keyword.  An additional way to set the pupil sampling interval for a particular set of phase
+    screens (i.e., for a particular PhaseScreenList) is to provide the screens in the `screen_list`
+    argument.  Each screen in the list computes its own preferred sampling rate and the
+    PhaseScreenList appropriately aggregates these. This last option also requires that a wavelength
+    `lam` be specified, and is particularly helpful for creating PSFs derived from turbulent
+    atmospheric screens.
+
+    Finally, when specifying the pupil geometrically, Aperture may choose to make a small adjustment
+    to `pupil_plane_scale` in order to produce an array with a good size for FFTs.  If your
+    application depends on knowing the size and scale used with the Fourier optics framework, you
+    can obtain these from the `Aperture.pupil_plane_size` and `Aperture.pupil_plane_scale`
+    attributes.
+
+    Pupil image specification
+    --------------------------
+
+    The second way to specify the pupil plane configuration is by passing in an image of it.  This
+    can be useful, for example, if the struts are not evenly spaced or are not radially directed, as
+    is assumed by the simple model for struts described above.  In this case, an exception is raised
+    if keywords related to struts are also given.  On the other hand, the `obscuration` keyword is
+    still used to ensure that the PSF images are not aliased, though it is ignored during the actual
+    construction of the pupil plane illumination pattern.  The `pupil_plane_im` that is passed in
+    can be rotated during internal calculations by specifying a `pupil_angle` keyword.  Also note
+    that given how the `obscuration` keyword is used, the obscuration in the supplied pupil plane
+    image should indeed be reasonably circular rather than having an arbitrary shape; however, minor
+    deviations from a perfect circle do not cause significant problems.
+
+    If you choose to pass in a pupil plane image, it must be a square array in which the image of
+    the pupil is centered.  The areas that are illuminated should have some value >0, and the other
+    areas should have a value of precisely zero.  Based on what the Aperture class determines is a
+    good PSF sampling interval, the image of the pupil plane that is passed in might be zero-padded
+    during internal calculations.  (The pupil plane array size and scale values can be accessed via
+    the `Aperture.pupil_plane_size` and `Aperture.pupil_plane_scale` attributes.) The pixel scale of
+    the pupil plane can be specified in one of three ways.  In descending order of priority, these
```

Comment:
I think this comment is addressed now.  I added a unit test that reproduced the bug #752 but now passes.